scipy/stats/_continuous_distns.py

```diff
@@ -5206,6 +5206,321 @@ def _entropy(self, beta):
+        """
+        norm_values, parameters = parameters[:len(self.distribution_norms)], parameters[len(self.distribution_norms):]
+        shape_values = []
+        for shape in self.distribution_shapes:
+            shape_values.append(parameters[:len(shape)])
+            parameters = parameters[len(shape):]
+        total_norm = np.sum(np.array(norm_values), axis=0)
+        norm_values = [norm_value / total_norm for norm_value in norm_values]
+        return norm_values, shape_values
+
+    def _pdf(self, x, *args):
+        """
+        The combined PDF is the sum of the distribution PDFs weighted by their individual norm factors
+        """
+        norm_values, shape_values = self._extract_positional_arguments(args)
+        return np.sum((norm * distribution.pdf(x, *shape) for norm, distribution, shape in zip(norm_values, self.distributions, shape_values)), axis=0)
+    
+    def _cdf(self, x, *args):
+        """
+        The combined CDF is the sum of the distribution CDFs weighted by their individual norm factors
+        """
+        norm_values, shape_values = self._extract_positional_arguments(args)
+        return np.sum((norm * distribution.cdf(x, *shape) for norm, distribution, shape in zip(norm_values, self.distributions, shape_values)), axis=0)
+    
+    def _rvs(self, *args):
+        """
+        Generates random numbers using the individual distribution random generator
+        with a probability given by their individual norm factors 
+        """
+        norm_values, shape_values = self._extract_positional_arguments(args)
+        choices = np.random.choice(len(norm_values), size=self._size, p=norm_values)
+        result = np.zeros(self._size)
+        for i, (distribution, shape) in enumerate(zip(self.distributions, shape_values)):
+            mask = choices == i
+            result[mask] = distribution.rvs(size=mask.sum(), *shape)
+        return result
+
+    def _updated_ctor_param(self):
+        """
+        Set distributions as additional constructor argument
+        """
+        dct = super(mixture_gen, self)._updated_ctor_param()
+        dct['distributions'] = self.ctor_argument
+        return dct
+
+    def _argcheck(self, *args):
+        """
+        We allow for arbitrary arguments.
+        The sub distributions will check their arguments later anyway.
+        The default _argcheck method restricts the arguments to be positive.
+        """
+        return 1
+
+
+class template_gen(rv_continuous):
+    """
+    Generates a distribution given by a histogram.
+    This is useful to generate a template distribution from a binned datasample.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    There are no additional shape parameters except for the loc and scale.
+    The pdf and cdf are defined as stepwise functions from the provided histogram.
+    In particular the cdf is not interpolated between bin boundaries and not differentiable.
```

Comment:
Also if this is added in some form, an instance must be added to the test loop in `test_continuous_basic.py`